pkg/cluster/internal/providers/docker/provider.go

```diff
@@ -39,6 +39,9 @@ import (
 	"sigs.k8s.io/kind/pkg/internal/sets"
 )
 
+// ProviderName is the constant provider name associated with the Docker provider.
+const ProviderName = "docker"
```

Comment:
We discussed this previously, we really don't want to litter the code with "if provider x do y, else do z", that is not sustainable with more providers.

Instead, we should expand the provider interface to surface the result we need.

Reply:
But also, I'm wondering why we can't use resolv.conf with docker as well ...